docs/source/system-metrics/index.rst

```diff
@@ -0,0 +1,214 @@
+
+Now starting an MLflow run will automatically collect and log the default system metrics. Try running the following code
+in your favorite environment, and you should see system metrics existing in the logged run data. Please note that you don't
+necessarilty need to start an MLflow server, as the metrics are logged locally.
+
+.. code-block:: python
+
+    import mlflow
+    import time
+
+    with mlflow.start_run() as run:
+        time.sleep(15)
+
+    print(mlflow.MlflowClient().get_run(run.info.run_id).data)
+
+Your output should look like this:
+
+.. code-block:: output
+
+    <RunData: metrics={'system/cpu_utilization_percentage': 12.4,
+    'system/disk_available_megabytes': 213744.0,
+    'system/disk_usage_megabytes': 28725.3,
+    'system/disk_usage_percentage': 11.8,
+    'system/network_receive_megabytes': 0.0,
+    'system/network_transmit_megabytes': 0.0,
+    'system/system_memory_usage_megabytes': 771.1,
+    'system/system_memory_usage_percentage': 5.7}, params={}, tags={'mlflow.runName': 'nimble-auk-61',
+    'mlflow.source.name': '/usr/local/lib/python3.10/dist-packages/colab_kernel_launcher.py',
+    'mlflow.source.type': 'LOCAL',
+    'mlflow.user': 'root'}>
+
+To disable system metrics logging, you can use either of the following commands:
+
+.. code-block:: bash
+
+    export MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING="false"
+
+.. code-block:: python
+
+    import os
+
+    del os.environ["MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING"]
+
+Rerunning the MLflow code above will not log system metrics.
+
+Use ``mlflow.enable_system_metrics_logging()``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We also provide a pair of APIs ``mlflow.enable_system_metrics_logging()`` and
+``mlflow.disable_system_metrics_logging()`` to turn on/off system metrics logging globally, since
+sometimes you won't be able to set the environment variable. Running the following code will have the
+same effect as setting ``MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING`` environment variable to ``true``:
+
+.. code-block:: python
+
+    import mlflow
+
+    mlflow.enable_system_metrics_logging()
+
+    with mlflow.start_run() as run:
+        time.sleep(15)
+
+    print(mlflow.MlflowClient().get_run(run.info.run_id).data)
+
+Turn on System Metrics Logging for a Single Run
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to controlling system metrics logging globally, you can also control it for a
+single run. To do so, set ``log_system_metrics`` as ``True`` or ``False`` accordingly in :py:func:`mlflow.start_run()`:
+
+.. code-block:: python
+
+    with mlflow.start_run(log_system_metrics=True) as run:
+        time.sleep(15)
+
+    print(mlflow.MlflowClient().get_run(run.info.run_id).data)
+
+The above code will log system metrics for the current run, even if you have disabled system metrics logging by setting
+``MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING`` to ``false`` or calling ``mlflow.disable_system_metrics_logging()``.
+
```

Comment:
Can we add a table that explains the order of precedence to having these 3 mechanism set so that users can know which one takes priority over others?

Reply:
good point, but it's a bit hard to make a table since method 1 and method 2 are of the same priority, only method 3 is of a higher priority.

I refactored the writeup to make it easier to understand, please take a look if it works.

Reply:
Yep! That makes it pretty clear :)